packages/amplify-category-notifications/src/pinpoint-helper.ts

```diff
@@ -0,0 +1,295 @@
+import {
+  $TSAny, $TSContext, $TSMeta, AmplifyCategories, AmplifySupportedService, open, stateManager,
+} from 'amplify-cli-core';
+
+import inquirer from 'inquirer';
+import ora from 'ora';
+
+import { printer } from 'amplify-prompts';
+import { deleteRolePolicy, ensureAuth } from './auth-helper';
+import { ICategoryMeta } from './notifications-types';
+
+const providerName = 'awscloudformation';
+const spinner = ora('');
+
+/**
+ * Get the Pinpoint app from analytics category
+ */
+export const getPinpointApp = (context: $TSContext): ICategoryMeta | undefined => {
+  const { amplifyMeta } = context.exeInfo;
+  const pinpointApp = scanCategoryMetaForPinpoint(amplifyMeta[AmplifyCategories.NOTIFICATIONS], undefined);
+  if (!pinpointApp) {
+    return scanCategoryMetaForPinpoint(amplifyMeta[AmplifyCategories.ANALYTICS], undefined);
+  }
+  return pinpointApp;
+};
+
+/**
+ * Ensure Pinpoint app exists
+ */
+export const ensurePinpointApp = async (context: $TSContext, pinpointNotificationsMeta: $TSMeta): Promise<void> => {
+  let pinpointApp;
+  let resourceName;
+  const { amplifyMeta, localEnvInfo } = context.exeInfo;
+
+  if (pinpointNotificationsMeta) {
+    if (
+      pinpointNotificationsMeta.service === AmplifySupportedService.PINPOINT
+      && pinpointNotificationsMeta.output
+      && pinpointNotificationsMeta.output.Id
```

Comment:
&& pinpointNotificationsMeta?.output?.Id

Reply:
this would make it different from the original JS file, looks that the check for output.id is done at line 39